.github/workflows/jsBuilder.yml

```diff
@@ -0,0 +1,42 @@
+name: Transpile TS
+
+on:
+  push:
+    branches: [ts-support-js-to-ts] # <-- Change this to the primary branch (for example, "dev" if you want all changes to `dev` to be watched).
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false # otherwise, the token used is the GITHUB_TOKEN, instead of your personal token.
+          fetch-depth: 0 # otherwise, you will failed to push refs to dest repo.
+
+      - name: Node setup
+        uses: actions/setup-node@v2
+        with:
+          node-version: '16'
+          cache: 'npm'
+
+      - name: Build JavaScript files
+        run: | # Change last line to your build script command #.
+          npm i
+          npm run start 
+
+      - name: Force add JS files to override .gitignore
+        run: git add --force ./lib # <-- Change this to your build path.
+
+      - name: Commit files
+        run: | # Change last line to your preferred commit message (I like `chore: build js files`).
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          git commit -m "Updating JS files" -a 
+
+      - name: Push changes
+        uses: ad-m/github-push-action@v0.6.0
+        with:
+          force: true
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          branch: ts-master # <-- Change this to the branch where you want builds to be (for example, "master" if you want changes to the primary branch to be applied to `master`).
```

Comment:
so basically this needs to be changed to master prior to merge?

Reply:
yes, this is changed to `master`